src/dataset-builder/ConceptSearch.ts

```diff
@@ -34,15 +34,17 @@ const DebouncedTextInput = withDebouncedChange(TextInput);
 export const ConceptSearch = (props: ConceptSearchProps) => {
   const { initialSearch, domainOption, onCancel, onCommit, onOpenHierarchy, actionText, datasetId, initialCart } =
     props;
-  const [search, setSearch] = useState<string>(initialSearch);
+  const [searchText, setSearchText] = useState<string>(initialSearch);
   const [cart, setCart] = useState<Concept[]>(initialCart);
   const [concepts, searchConcepts] = useLoadedData<GetConceptsResponse>();
 
   useEffect(() => {
-    void searchConcepts(() => {
-      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, search);
-    });
-  }, [search, datasetId, domainOption.root, searchConcepts]);
+    if (searchText.length === 0 || searchText.length > 2) {
```

Comment:
If you wanted to, you could add a test that verifies that no search occurs if the text is one or two characters.